serverless/networking/serverless-ossm-tls.adoc

```diff
@@ -0,0 +1,165 @@
+include::modules/serverless-document-attributes.adoc[]
+include::modules/ossm-document-attributes.adoc[]
+[id="serverless-ossm-tls"]
+= Configuring TLS for a custom domain using {ProductName}
+:context: serverless-ossm-tls
+include::modules/common-attributes.adoc[]
+
+toc::[]
+
+.Prerequisites
+* Install xref:../../serverless/installing_serverless/installing-openshift-serverless.adoc#installing-openshift-serverless[{ServerlessProductName}].
+* Install xref:../../service_mesh/service_mesh_install/preparing-ossm-installation.adoc#preparing-ossm-installation[{ProductName}].
+* In this example, `openssl` is used to generate certificates, but you can use any certificate generation tool you like to create these.
+
+.Procedure
+
+. Create TLS certificates:
++
+
+[source,terminal]
+----
+$ openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj '/O=Example Inc./CN=example.com' -keyout example.com.key -out example.com.crt
+----
+
++
+
+[source,terminal]
+----
+$ openssl req -out custom.example.com.csr -newkey rsa:2048 -nodes -keyout custom.example.com.key -subj "/CN=custom-ksvc-domain.example.com/O=Example Inc."
+----
+
++
+
+[source,terminal]
+----
+$ openssl x509 -req -days 365 -CA example.com.crt -CAkey example.com.key -set_serial 0 -in custom.example.com.csr -out custom.example.com.crt
+----
+
+. Check that the certificates appear in your directory:
++
+
+[source,terminal]
+----
+$ ls -1
+----
+
++
+
+.Example output
+[source,terminal]
+----
+custom.example.com.crt
+custom.example.com.csr
+custom.example.com.key
+example.com.crt
+example.com.key
+----
+
+. Create a secret:
++
+
+[source,terminal]
+----
+$  oc create -n istio-system secret tls custom.example.com --key=custom.example.com.key --cert=custom.example.com.crt
+----
+
+. Attach the secret to the Istio ingress gateway, by editing the `ServiceMeshControlPlane` resource.
+.. Edit the `ServiceMeshControlPlane` resource:
++
+
+[source,terminal]
+----
+$ oc edit -n istio-system ServiceMeshControlPlane <control_plane_name>
+----
+
+.. Check that the following lines exist in the resource, and if they do not, add them:
++
+
+[source,yaml]
+----
+istio-ingressgateway:
+  secretVolumes:
+  - mountPath: /custom.example.com
+    name: custom-example-com
+    secretName: custom.example.com
+----
+
+. Update the Istio ingress gateway to use your secret.
+.. Edit the `default-gateway` resource:
++
+
+[source,terminal]
+----
+$ oc edit gateway default-gateway
```

Comment:
I've installed Serverless with Serving, and Service mesh with dependencies. Added SMCP and namespaces to SMMR.

Previous command worked well, but this refers to `default-gateway` but there aren't any:

```
$ oc get gateway -A
No resources found
```

I saw that this gateway is defined in different document in paragraph [Creating necessary Service Mesh resources](https://tls--ocpdocs.netlify.app/openshift-enterprise/latest/serverless/networking/serverless-ossm-custom-domains.html#serverless-service-mesh-resources_serverless-ossm-custom-domains)

Reply:
Yeah, we should add "Follow `Using custom domains for Knative services with Service Mesh` section` to the `Prerequisites`.

This section is `Configuring TLS for a custom domain using Red Hat OpenShift Service Mesh` so we need to follow the [custom domain doc](https://tls--ocpdocs.netlify.app/openshift-enterprise/latest/serverless/networking/serverless-ossm-custom-domains.html) first.

Reply:
Added